flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/DisplayUnit.js

```diff
@@ -0,0 +1,67 @@
+class DisplayUnit {
+    constructor(isPoweredFn, getSelfTestTimeInSecondsFn, potentiometerId, selfTestElement) {
+        this.selfTest = new DisplayUnitSelfTest(selfTestElement, getSelfTestTimeInSecondsFn);
+        this.isPowered = isPoweredFn;
+        this.potentiometerId = potentiometerId;
+
+        this.previouslyOff = false;
+    }
+
+    isJustNowTurnedOn() {
+        return this.previouslyOff && this.isOn();
+    }
+
+    isOn() {
+        const brightness = SimVar.GetSimVarValue(`LIGHT POTENTIOMETER:${this.potentiometerId}`, "number");
+        return brightness >= 0.1 && this.isPowered();
+    }
+
+    update(deltaTime) {
+        if (this.isJustNowTurnedOn()) {
+            this.selfTest.execute();
+        }
+
+        const isOn = this.isOn();
+        this.selfTest.update(deltaTime, isOn);
+
+        this.previouslyOff = !isOn;
+    }
+}
+
+class DisplayUnitSelfTest {
+    constructor(element, getSelfTestTimeInSecondsFn) {
+        this.element = element;
+        this.getSelfTestTimeInSeconds = getSelfTestTimeInSecondsFn;
+
+        this.remainingTestDurationInMilliseconds = 0;
+
+        // Start with a state where turning on the display unit within 10 seconds after starting the flight
+        // will trigger the self test.
+        this.offDurationInMilliseconds = DisplayUnitSelfTest.RequiredAfterBeingOffForMilliseconds;
+    }
+
+    static get RequiredAfterBeingOffForMilliseconds() {
```

Comment:
`powerOffRideThroughTime` or something like that might be clearer, or at least relate to the physical state being modelled?

IMO the logic for determining whether the caps are fully discharged/the DU is fully dead belongs in the DisplayUnit, and the SelfTest should simply be concerned with doing the self test.

Reply:
> `powerOffRideThroughTime` or something like that might be clearer, or at least relate to the physical state being modelled?

I guess it might be clearer for some and less clear for others. I myself didn't know ride through time. It seems to be an electrical engineering term.

> IMO the logic for determining whether the caps are fully discharged/the DU is fully dead belongs in the DisplayUnit, and the SelfTest should simply be concerned with doing the self test.

Yes. In a previous version I actually programmed the DU to contain the duration of being off and I agree with you that from the perspective of "what object has which properties" it makes most sense to put it in the DU. However, as the self test is the only piece of the DU that cares about the duration the DU was off (at least at this moment), I decided to encapsulate it in there as to not bother the higher-level abstraction DU with it.

So now the question is, what do we prefer? 😃

Reply:
> I guess it might be clearer for some and less clear for others. I myself didn't know ride through time. It seems to be an electrical engineering term.

Fair point!